kratos/processes/find_intersected_geometrical_objects_process.cpp

```diff
@@ -178,19 +178,17 @@ void FindIntersectedGeometricalObjectsProcess::Execute()
     // Calling initialize first (initialize Octree)
     ExecuteInitialize();
 
-    OtreeCellVectorType leaves;
-
     // Iterate over elements
     if (mOptions.Is(FindIntersectedGeometricalObjectsProcess::INTERSECTED_ELEMENTS)) {
         auto& r_elements_array = mrModelPartIntersected.Elements();
         const SizeType number_of_elements = r_elements_array.size();
 
         const auto it_elements_begin = r_elements_array.begin();
 
-        IndexPartition<std::size_t>(number_of_elements).for_each(leaves ,[&](std::size_t index, OtreeCellVectorType& local_leaves){
+        IndexPartition<std::size_t>(number_of_elements).for_each(OtreeCellVectorType() ,[&](std::size_t index, OtreeCellVectorType& rLocalLeaves){
             auto it_elem = it_elements_begin + index;
-            leaves.clear();
-            IdentifyNearEntitiesAndCheckEntityForIntersection(*(it_elem.base()), local_leaves);
+            rLocalLeaves.clear();
```

Comment:
Do we need clear() here??? Anyways they are new and local right ? @philbucher

Reply:
well they are shared in this tread so I think the clear is necessary